packager/media/codecs/ec3_audio_util.cc

```diff
@@ -121,8 +189,20 @@ bool ExtractEc3Data(const std::vector<uint8_t>& ec3_data,
   *dependent_substreams_layout = 0;
   if (number_dependent_substreams > 0) {
     RCHECK(bit_reader.ReadBits(9, dependent_substreams_layout));
+  } else {
+    RCHECK(bit_reader.SkipBits(1));
+  }
+  if (bit_reader.bits_available() < 16) {
+    *ec3_joc_flag = false;
+    *ec3_joc_complexity = 0;
+    return true;
   }
 
+  RCHECK(bit_reader.SkipBits(7));
+  RCHECK(bit_reader.ReadBits(1, ec3_joc_flag));
+  if (*ec3_joc_flag == true) {
```

Comment:
nit: you can remove "== true"

And a question: do we need two variables? Is it ok to use ec3_joc_complexity==0 to indicate ec3_joc_flag=false?

Reply:
nit: you can remove "== true"

- Yes, I will remove.

And a question: do we need two variables? Is it ok to use ec3_joc_complexity==0 to indicate ec3_joc_flag=false?

- Yes. We need two variables here.  
Comparing with regular DD+ stream, it contains additional 2 (or more) bytes. 
The syntax of the first byte is **reserved bits (6 bits)** + **flag_ec3_extension_type_b (1 bit)** + **flag_ec3_extension_type_a (1 bit)**. 
Only if flag_ec3_extension_type_a == 1, the second byte represent ec-3 joc complexity.
If flag_ec3_extension_type_b == 1, the second byte will have another meaning.
flag_ec3_extension_type_a  and flag_ec3_extension_type_b can't be set to 1 simultaneously.

Reply:
ec3_joc_complexity==0 can be used to indicate indicate ec3_joc_flag=false.
But ec3_joc_complexity !=0 can NOT be used to indicate ec3_joc_flag=ture.

Reply:
I think there are some misunderstandings. What I mean is that can we keep `ec3_joc_flag` as a local variable and just expose |ec3_joc_complexity| to the caller, i.e.

```c++
  bool ec3_joc_flag;
  RCHECK(bit_reader.ReadBits(1, &ec3_joc_flag));
  if (ec3_joc_flag) {
    ...
  }
```

And in DASH and HLS generation, the code can check whether |ec3_joc_complexity| is 0 first before generating the corresponding DASH/HLS tags.

This works only if 0 is not a valid value for |ec3_joc_complexity|, so we can use it to indicate the presence of the flag.

This allows us to remove |ec3_joc_flag| in the proto. (Actually we can already do that now by using protobuf's has_xxx function to determine the presence of the flag)

Reply:
Let me have a discussion with team firstly, and then give you feedback late.

Reply:
We can remove the ec3_joc_flag. I have updated the PR following the suggestion. Please have a review.